File: goty-server/src/main/java/com/aleinin/goty/GameOfTheYearSubmission.java
package com.aleinin.goty;

import lombok.Value;
import lombok.With;
import org.springframework.data.annotation.Id;

import java.util.List;
import java.util.UUID;

@Value
public class GameOfTheYearSubmission {

    @Id
    @With
    UUID id;
    String name;
    List<RankedGameSubmission> gameSubmissions;
    GameSubmission mostAnticipated;
    GameSubmission bestOldGame;

    @Value
    public static class GameSubmission {

        String id;
        String title;
    }

    @Value
    public static class RankedGameSubmission {

        String id;
        String title;
        int rank;
    }
}

File: goty-server/src/main/java/com/aleinin/goty/submit/GameOfTheYearSubmission.kt
package com.aleinin.goty.submit

import org.springframework.data.annotation.Id
import java.util.*

data class GameOfTheYearSubmission(
        @Id val id: UUID,
        val name: String,
        val gameSubmissions: List<RankedGameSubmission>,
        val mostAnticipated: GameSubmission,
        val bestOldGame: GameSubmission) {

    data class GameSubmission(
            val id: String,
            val title: String)

    data class RankedGameSubmission(
            val id: String,
            val title: String,
            val rank: Int)
}

